src/Raven.Server/TrafficWatch/TrafficWatchToLog.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Raven.Client.Documents.Changes;
+using Raven.Client.ServerWide.Operations.TrafficWatch;
+using Raven.Client.Util;
+using Raven.Server.Config.Categories;
+using Sparrow;
+using Sparrow.Logging;
+using Size = Sparrow.Size;
+
+
+namespace Raven.Server.TrafficWatch;
+
+internal class TrafficWatchToLog
+{
+    private static readonly Logger Logger = LoggingSource.Instance.GetLogger<RavenServerStartup>("TrafficWatchManager");
+
+    public TrafficWatchConfiguration Configuration { get; private set; } = new();
+
+    public bool LogToFile => Configuration.TrafficWatchMode == TrafficWatchMode.ToLogFile || Logger.IsOperationsEnabled;
+
+    private TrafficWatchToLog() { }
+
+    public static TrafficWatchToLog Instance = new TrafficWatchToLog();
+
+    public void Log(TrafficWatchChangeBase trafficWatchData)
+    {
+        string msg = null;
+
+        if (Logger.IsOperationsEnabled == false)
+            return;
+
+        if (Configuration.TrafficWatchMode == TrafficWatchMode.Off)
+            return;
+
+        if (trafficWatchData is TrafficWatchHttpChange twhc)
+        {
+            if (Configuration.Databases != null)
+                if (Configuration.Databases.Count > 0 &&
+                    Configuration.Databases.Contains(twhc.DatabaseName) == false)
+                    return;
+
+            if (Configuration.HttpMethods != null)
+                if (Configuration.HttpMethods.Count > 0 &&
+                    Configuration.HttpMethods.Contains(twhc.HttpMethod) == false)
+                    return;
+
+            if (Configuration.ChangeTypes != null)
+                if (Configuration.ChangeTypes.Count > 0 &&
+                    Configuration.ChangeTypes.Contains(twhc.Type) == false)
+                    return;
+
+            if (Configuration.StatusCodes != null)
+                if (Configuration.StatusCodes.Count > 0 &&
+                    Configuration.StatusCodes.Contains(twhc.ResponseStatusCode) == false)
+                    return;
+
+            if (Configuration.MinimumResponseSize.GetValue(SizeUnit.Bytes) > twhc.ResponseSizeInBytes)
+                return;
+
+            if (Configuration.MinimumRequestSize.GetValue(SizeUnit.Bytes) > twhc.RequestSizeInBytes)
+                return;
+
+            if (Configuration.MinimumDuration > twhc.ElapsedMilliseconds)
+                return;
+
+            var requestSize = new Size(twhc.RequestSizeInBytes, SizeUnit.Bytes);
+            var responseSize = new Size(twhc.ResponseSizeInBytes, SizeUnit.Bytes);
+            var customInfo = twhc.CustomInfo?.ReplaceLineEndings("") ?? "N/A";
+
+            msg = $"HTTP, {twhc.TimeStamp:MM/dd/yyyy HH:mm:ss}, {twhc.DatabaseName}, " +
```

Comment:
is it enough to have time precision to seconds? not milliseconds? in the studio we have:

![image](https://user-images.githubusercontent.com/376551/185361987-a75e4065-cb2e-4c0d-bf64-7837b734551a.png)

Reply:
I did what the [issue](https://issues.hibernatingrhinos.com/issue/RavenDB-17037/Traffic-watch-log-to-the-log-file) assigned.
In fact, I think we need to remove this duplicate time in this place since time will be taken by the `Logger.Operations`. I hope @grisha-kotler and @haludi tell us exactly how it should be.

Reply:
If it is the same time there is no need to duplicate it.

Reply:
Duplicated timestamp removed.